pkg/api/devops/v1alpha1/template_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TemplateSpec defines the desired state of Template
+type TemplateSpec struct {
+	// Parameters are used to configure template.
+	//+optional
+	Parameters []TemplateParameter `json:"parameters,omitempty"`
+
+	// Template is a string with go-template style.
+	Template string `json:"template,omitempty"`
+}
+
+// TemplateStatus defines the observed state of Template
+type TemplateStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// TemplateParameter is definition of how can we configure our parameter.
+type TemplateParameter struct {
+	// Name is name of the parameter.
+	Name string `json:"name"`
+
+	// Description is description of the parameter.
+	//+optional
+	Description string `json:"description,omitempty"`
+
+	// Default is default value of the parameter.
+	//+optional
+	Default apiextensionv1.JSON `json:"default,omitempty"`
```

Comment:
why use `apiextensionv1.JSON` instead of `string`?

Reply:
The type of the default value might be any type, not only string, so I defined it to JSON which is exact byte array. You can refer to [here](https://github.com/tektoncd/triggers/blob/c8ca0ed2d0e5a4a8063501058055a09b931062ed/pkg/apis/triggers/v1alpha1/trigger_types.go#L111-L114).